include/sampleflow/producers/delayed_rejection_mh.h

```diff
@@ -165,26 +164,29 @@ namespace SampleFlow
               OutputType trial_sample = std::move(trial_sample_and_ratio.first);
               const double proposal_distribution_ratio = trial_sample_and_ratio.second;
               const double trial_log_likelihood = log_likelihood (trial_sample);
+              const double likelihood_ratio = std::exp(trial_log_likelihood - current_log_likelihood);
               // Accept trial sample with probability equal to ratio of likelihoods;
               // (always accept if > 1)
-              double acceptance_ratio = (std::exp(trial_log_likelihood - current_log_likelihood) /
-                                         proposal_distribution_ratio) * prev_acceptance_ratio;
-              double u = uniform_distribution(rng);
-              if (acceptance_ratio > 1 || acceptance_ratio >= u)
+              double orig_acceptance_ratio = likelihood_ratio * proposal_distribution_ratio;
+              double acceptance_ratio = orig_acceptance_ratio;
+              acceptance_ratios.push_back(orig_acceptance_ratio)
+
+              for (int j = 0; j < delay_stage; ++j)
+                {
+                  acceptance_ratio *= ((1 - acceptance_ratios[j + 1]) / (1 - acceptance_ratios[j]));
```

Comment:
It's hard to follow whether this is the correct formula. I think it would greatly help if you added the formula you're trying to implement as a comment, with the formula being what's listed in the paper.